### PR TITLE
Rewrite to a shared GitHub Actions workflow

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

**Review by ofedoren — COMMENTED** [review]
What should we do about https://github.com/theforeman/foreman_discovery/pull/612/files#diff-5bf32edc05219804403641dadf678cb09f975f656b06b6a199c1c737ebaf34ebR11 ? Mention it explicitly or wait for https://github.com/theforeman/actions/pull/16?

Otherwise LGTM and looks like this is the way now/near future.

**Review by ekohl — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,71 +1,19 @@
...
-      - uses: actions/checkout@v2
-      - name: Setup Ruby
-        uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: 2.7
-          bundler-cache: true
-      - name: Run rubocop
-        run: bundle exec rubocop
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
```
Comment (ekohl) [inline .github/workflows/ci.yml:12]: You can either declare the task in `Rakefile`, or override the command now:
```suggestion
    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
    with:
      command: bundle exec rubocop --parallel --format github
```

Comment (ekohl) [reply .github/workflows/ci.yml:12]: Sorry, wrong repo. This isn't an issue here.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,71 +1,19 @@
 name: CI
-on: [push, pull_request]
-env:
-  RAILS_ENV: test
-  DATABASE_URL: postgresql://postgres:@localhost/test
-  DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL: true
-  BUNDLE_WITHOUT: "journald:development:console:mysql2:sqlite"
+
+on: pull_request
```
Comment (ekohl) [inline .github/workflows/ci.yml:3]: Btw, it's also fine to keep building on push (like we used to). That way you get a shared cache for future PRs. You can consider limiting it to just master and any stable branches.

Comment (ofedoren) [reply .github/workflows/ci.yml:3]: Let's add on push to at least mater branch, wouldn't hurt I guess.

**Review by ofedoren — APPROVED** [review]
Thanks, @adamruzicka and @ekohl !

**Pull request merged**